avro-serde/src/main/java/io/confluent/kafka/streams/serdes/GenericAvroDeserializer.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2017 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.confluent.kafka.streams.serdes;
+
+import org.apache.avro.generic.GenericRecord;
+import org.apache.kafka.common.serialization.Deserializer;
+
+import java.util.Map;
+
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
+import io.confluent.kafka.serializers.KafkaAvroDeserializer;
+
+public class GenericAvroDeserializer implements Deserializer<GenericRecord> {
+
+  private final KafkaAvroDeserializer inner;
+
+  /**
+   * Implementation detail: This constructor is used by Kafka's Streams API.
+   */
+  public GenericAvroDeserializer() {
+    inner = new KafkaAvroDeserializer();
+  }
+
+  public GenericAvroDeserializer(final SchemaRegistryClient client) {
```

Comment:
Should this and the following be package private? Is there a good reason they need to be public APIs? I think we have them just because the KafkaAvroSerializer/Deserializer have similar ones but I'm not sure they are well motivated.

The second one, in particular, I think was a definite mistake (and my mistake when adding the deserializer) -- I don't think there's any good reason to have that version just to elide a call to `configure()`.